### DISCUSSION TITLE
New Sublime Text 4 packages to work with Prisma !

### DISCUSSION THREAD

**Discussion by UltraInstinct05** [prisma/prisma#12316] [Q&A]
Hi everyone 👋 

Today, I have released 2 new Sublime Text 4 packages to make working with Prisma schema files, a lot easier.

1. [PrismaHighlight](https://github.com/Sublime-Instincts/PrismaHighlight) - An enhanced syntax highlighting package for prisma schema files.

2. [LSP-prisma](https://github.com/Sublime-Instincts/LSP-prisma) - `LSP-prisma` is a `LSP` helper package which uses the [Prisma Language Tools](https://github.com/prisma/language-tools/tree/main/packages/language-server) and the main [LSP package](https://packagecontrol.io/packages/LSP) to provide linting, formatting capabilities, rich auto completion & hover documentation capability for ST using the `LSP` protocol.

So, `PrismaHighlight` + `LSP` + `LSP-prisma` together should make for an ideal setup to work with Prisma schema files in Sublime Text 4.

If you want to use Sublime Text 4 to work with Prisma, then do give these packages a try ! As usual, happy to answer any questions ! 🙂 

Ashwin Shenoy

**Comment by Jolg42** [comment]
Just tried it and congrats! It's super nice! 💚 

Happy to add it to the docs here https://www.prisma.io/docs/guides/development-environment/editor-setup#sublime-text

Maybe after both packages are published to Package Control, because it will be easier for people to install?
Progress tracked in:
https://github.com/Sublime-Instincts/LSP-prisma/issues/1
https://github.com/Sublime-Instincts/PrismaHighlight/issues/1

Note: the previous Sublime Text 3 plugin, actually also works on v4 but doesn't have any LSP features
https://github.com/prisma/sublimeText3

**Reply by UltraInstinct05** [reply]
Thanks for the kind words !

> Maybe after both packages are published to Package Control, because it will be easier for people to install?

Yes, I have no issues. The installation steps provided should still be fairly simple, but adding it to PC will be ideal and I plan to do it soon. `LSP-prisma` shouldn't be a problem because there is no other package like it that provides intellisense for Prisma schema files, but for `PrismaHighlight`, there is already the aforementioned ST3 package https://packagecontrol.io/packages/Prisma. What I could do is restrict the ST3 package to ST3 & ST <4126 and `PrismaHighlight` to ST4, so that everything continues to work for people using ST3 as well.

> Note: the previous Sublime Text 3 plugin, actually also works on v4 but doesn't have any LSP features

It will continue to work because the `.tmLanguage` support continues in ST4 for backwards compatibility. But the issue with this format is that it's a legacy syntax format (at least in the ST world) and it won't receive any new features (like extending syntaxes, branching for multiline constructs, ability to debug syntaxes etc.) compared to the recommended `.sublime-syntax` format.

Also thanks for all the efforts put into the `Prisma Language Tools`. `LSP-prisma` is basically useless without it. And any enhancements to the language tools can automatically be enjoyed by `LSP-prisma` & ST users in the future (just like VS Code).

**Reply by Jolg42** [reply]
>What I could do is restrict the ST3 package to ST3 & ST <4126 and PrismaHighlight to ST4, so that everything continues to work for people using ST3 as well.

We could also not restrict anything and both Packages would be available to choose.

Though I like your idea since the Prisma ST3 Package in unmaintained.


>And any enhancements to the language tools can automatically be enjoyed by LSP-prisma & ST users in the future (just like VS Code).

Yes that's magic 💯

**Reply by UltraInstinct05** [reply]
> We could also not restrict anything and both Packages would be available to choose.

We could, but the review process for submitting packages to Package Control is kind of strict, gated behind community review before acceptance (Having participated & reviewed a number of submissions myself). I don't know what the process is with VS Code (meaning, can anyone can submit any package ?), but in case of PC, at least the question would be the necessity of 2 highlighting packages for the same thing. Hence the suggestion above.

**Reply by Jolg42** [reply]
Oh ok! I'm not familiar with that process, if that's needed to make it happen then I can approve that 👍🏼 

For VS Code there is no gatekeeper, it's free buffet 😄

**Reply by Jolg42** [reply]
It could make sense to make the Prisma ST3 Package restriction to <=3 and PrismaHighlight >=4 ?

**Reply by UltraInstinct05** [reply]
> It could make sense to make the Prisma ST3 Package restriction to <=3 and PrismaHighlight >=4 ?

Yes, this is the best option. Even though ST3 is now technically obsolete (meaning it won't receive _any_ updates), this approach has the best backwards compatibility, won't break for the users already using the ST3 package on ST3 and ST4 users can receive the updates from the new package. So, I'll make the submission with this in mind.

Also thanks for all the help & support !

**Comment by UltraInstinct05** [comment]
Alright, both packages are now submitted for acceptance.

1. `PrismaHighlight` - https://github.com/wbond/package_control_channel/pull/8498
2. `LSP-prisma` - https://github.com/sublimelsp/repository/pull/62

Once these PR's are merged, both of the packages will be available for install using PC the regular way.

Since all of this is done by the community, it may take a while for them to merge the PR's. Till then, of course, the alternate installation instructions in the `README` for both packages should work for anyone (which should also be simple).

Have fun using Prisma with ST ❤️

**Reply by Jolg42** [reply]
Approved both there 💚

**Comment by UltraInstinct05** [comment]
As an update on this topic, both packages are now available in Package Control.

https://packagecontrol.io/packages/Prisma
https://packagecontrol.io/packages/LSP-prisma

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
